Resources/ProxyProvider/PUSPProxyProvider.py

```diff
@@ -18,49 +18,35 @@ def __init__(self, parameters=None):
 
     super(PUSPProxyProvider, self).__init__(parameters)
 
-  def getProxy(self, userDict):
-    """ Generate user proxy
+  def checkStatus(self, userDN):
+    """ Read ready to work status of proxy provider
 
-        :param dict userDict: user description dictionary with possible fields:
-               FullName, UserName, DN, Email, DiracGroup
+        :param str userDN: user DN
 
-        :return: S_OK(basestring)/S_ERROR() -- basestring is a proxy string
+        :return: S_OK(dict)/S_ERROR() -- dictionary contain fields:
+                  - 'Status' with ready to work status[ready, needToAuth]
     """
+    if not userDN.split(":")[-1]:
+      return S_ERROR('Can not found user label for DN: %s' % userDN)
+    if not self.parameters.get('ServiceURL'):
+      return S_ERROR('Can not determine PUSP service URL')
+
+    return S_OK({'Status': 'ready'})
 
-    userDN = userDict.get('DN')
-    if not userDN:
-      return S_ERROR('Incomplete user information')
+  def getProxy(self, userDN):
+    """ Generate user proxy
 
-    diracGroup = userDict.get('DiracGroup')
-    if not diracGroup:
-      return S_ERROR('Incomplete user information')
+        :param str userDN: user DN
 
-    result = Registry.getGroupsForDN(userDN)
+        :return: S_OK(dict)/S_ERROR() -- dict contain 'proxy' field with is a proxy string
```

Comment:
I also don't see why `S_OK({'proxy': 'proxyString'})` is returned, instead of just `S_OK('proxyString')`

Reply:
Yes, I think it makes sense.